assets/src/data/eventespresso/schema/resolvers.js

```diff
@@ -45,19 +62,66 @@ export function* getFactoryForModel( modelName, schema = {} ) {
 }
 
 /**
- * A control for retrieving the schema for the given model
+ * A resolver for getting the relation endpoint for a given model, it's id, and
+ * the requested relation.
+ *
+ * The EE REST api names relations according to whether they there are singular
+ * or plural relations on a given model (eg. Registrations have one event
+ * relation, but Events can have multiple datetimes).  This means the only way
+ * to derive an accurate endpoint for a given relation request on an entity is
+ * to retrieve the entity from the resource and derive the endpoint from the
+ * links in the response.
+ *
+ *
  * @param {string} modelName
- * @return {IterableIterator<*>|Object}  a generator or Object if schema is
- * retrieved.
+ * @param {number} entityId
+ * @param {string} relationModelName
+ * @return {IterableIterator<*>|string} A generator or the derived endpoint.
  */
-function* getSchemaByModel( modelName ) {
-	let schema;
-	const resolved = yield select( 'hasResolvedSchemaForModel', modelName );
-	if ( resolved === true ) {
-		schema = yield select( 'getSchemaForModel', modelName );
-		return schema;
+export function* getRelationEndpointForEntityId(
+	modelName,
+	entityId,
+	relationModelName
+) {
+	// first attempt to get the relation endpoint from the entity that might
+	// already be in core state.
+	const entity = yield select(
+		CORE_REDUCER_KEY,
+		'getEntityById',
+		modelName,
+		entityId
+	);
+	const pluralRelationName = pluralModelName( relationModelName );
+	let endpoint = '';
+	if ( isModelEntity( entity ) && entity[ pluralRelationName + 'Resource' ] ) {
+		endpoint = stripBaseRouteFromUrl(
+			entity[ pluralRelationName + 'Resource' ].resourceLink
+		);
+	} else {
+		const response = yield fetchFromApi(
+			{
+				path: getEndpoint( modelName ) + '/' + entityId,
+			}
+		);
+		const links = response._links || {};
+		if ( ! links ) {
+			return '';
+		}
+		const baseRelationPath = 'https://api.eventespresso.com/';
```

Comment:
what??? this can't be right?

if you go to that URL, you will see:

- that that URL is not actually HTTPS and your browsers throws a security tantrum
- there's a WP site named "Brent's Testing Site" (which I do not remember setting up)

why is this not relative to the site running this code?